core/src/test/java/org/apache/calcite/test/SqlToRelConverterTest.java

```diff
@@ -4769,4 +4769,18 @@ void checkUserDefinedOrderByOver(NullCollation nullCollation) {
     String sql = "SELECT CAST(CAST(? AS INTEGER) AS CHAR)";
     sql(sql).ok();
   }
+
+  /**
+   * Test case for
+   * <a href="https://issues.apache.org/jira/browse/CALCITE-5468">[CALCITE-5468]
+   * SqlToRelConverter should register sub-queries inside ORDER BY clause
+   * for queries without aggregation</a>.
+   */
+  @Test void testOrderByWithSubQuery() {
+    String sql = "SELECT empno\n"
+        + "FROM emp\n"
+        + "ORDER BY\n"
+        + "CASE WHEN empno IN (1,2) THEN 0 ELSE 1 END";
+    sql(sql).ok();
+  }
```

Comment:
Don't put stuff on the end. It causes merge conflicts. Put it in the most logical place.

Reply:
This IN is not a subquery. Can you rename this test, and add a test for a subquery?

Reply:
Yes, IN is not subquery in terms of SQL. But inside codebase there are places with code like:
```
/**
 * Scalar expression that represents an IN, EXISTS or scalar sub-query.
 */
public class RexSubQuery extends RexCall
```
It is actually quite confusing.

I renamed test and added test for subquery.